pkg/test/toolchain/tekton_assertion.go

```diff
@@ -40,10 +45,42 @@ func AssertThatTektonInstallation(t *testing.T, ns, name string, client client.C
 	}
 }
 
+// HasOwnerRef verifies that the Tekton installation has the expected ownerReference
+func (a *TektonInstallationAssertion) HasOwnerRef(sub *opsv1alpha1.Subscription) *TektonInstallationAssertion {
+	err := a.loadTektonInstallationAssertion()
+	require.NoError(a.t, err)
+
+	err = a.client.Get(context.TODO(), types.NamespacedName{Namespace: sub.GetNamespace(), Name: sub.GetName()}, sub)
+	require.NoError(a.t, err)
+
+	references := a.tektonInstallation.ObjectMeta.GetOwnerReferences()
```

Comment:
we could perhaps use the `runtime.Unstructured` type to get the ObjectMeta without having to duplicate these `HasOwnerRef` and `HasNoOwnerRef` funcs.

Reply:
Then we would need to have some higher hierarchy of the Assertion types - for now, they are separated - there is no interface nor composition so I need to "duplicate" it. I at least moved the Get call to the common assertion, but that's all I can do to minimize duplication: 199f6c9

Reply:
well, instead of making a larger refactoring of the Assertions, I thought we could have a more generic `hasOwnerRef` (private) func which would be called by the 2 `HasOwnerRef` methods of the Assertions.

Reply:
But still, what would we safe here? Currently, the function looks like this:
```go
func (a *TektonInstallationAssertion) HasOwnerRef(sub *opsv1alpha1.Subscription) *TektonInstallationAssertion {
	err := a.loadTektonInstallationAssertion()
	require.NoError(a.t, err)

	references := a.tektonInstallation.ObjectMeta.GetOwnerReferences()
	assertThatContainsOwnerReference(a.t, a.client, references, sub)
	return a
}
```
We need to call the first two lines anyway to load the TektonInstallation and check that was found. So we could only move 
`references := a.tektonInstallation.ObjectMeta.GetOwnerReferences()` 
inside of the generic function, but for what price? We would need to cast it to `runtime.Unstructured` and retrieve the metadata inside of the generic function... 
If you don't like the additional line, then I can do this:
`assertThatContainsOwnerReference(a.t, a.client, a.tektonInstallation.ObjectMeta, sub)`
and retrieve the references inside of the generic function.

Reply:
right, never mind, it is probably over-engineering...